game-core/src/main/java/games/strategy/triplea/delegate/BattleExtendedDelegateState.java

```diff
@@ -10,11 +10,14 @@
   BattleTracker battleTracker = new BattleTracker();
   public boolean needToInitialize;
   boolean needToScramble;
+  boolean needToCreateRockets;
   boolean needToKamikazeSuicideAttacks;
   boolean needToClearEmptyAirBattleAttacks;
   boolean needToAddBombardmentSources;
+  boolean needToFireRockets;
   boolean needToRecordBattleStatistics;
   boolean needToCheckDefendingPlanesCanLand;
   boolean needToCleanup;
+  RocketsFireHelper rocketHelper;
```

Comment:
Saving RocketsFireHelper in the state isn't really ideal here.

Reply:
To give a longer answer, if you insisted on saving the state information somewhere else, that would mean we would need to break encapsulation to allow the state information to be copied. I don't understand this idea. Isn't encapsulation one of the main pluses of object orientated programming?

Reply:
From a users perspective, if they do a rocket attack and an SBR with a potential intercept, then save the game at the point of the intercept question which (without the property being true) means that the rocket target selection has already been done, the save game then should include the rocket targets which have been selected.

Hopefully, that gives you enough information to not need any follow up questions.